File: 백준/Silver/18258. 큐 2/큐 2.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.Deque;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Deque<Integer> deque = new ArrayDeque<>();

        int N = Integer.parseInt(br.readLine());

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String command = st.nextToken();

            switch (command) {
                case "push":
                    int value = Integer.parseInt(st.nextToken());
                    deque.add(value);
                    break;
                case "pop":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.poll() + "\n");
                    }
                    break;
                case "size":
                    bw.write(deque.size() + "\n");
                    break;
                case "empty":
                    bw.write((deque.isEmpty() ? 1 : 0) + "\n");
                    break;
                case "front":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.peek() + "\n");
                    }
                    break;
                case "back":
                    if (deque.isEmpty()) {
                        bw.write("-1\n");
                    } else {
                        bw.write(deque.peekLast() + "\n");
                    }
                    break;
            }
        }
        bw.flush();
        bw.close();
        br.close();
    }
}

File: 백준/Silver/18258. 큐 2/큐 2.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val sb = StringBuilder() // 결과 저장
    val que = LinkedList<Int>() // 큐를 구현할 연결 리스트
    val n = br.readLine().toInt() // 입력 수

    repeat(n) {
        val command = br.readLine().split(" ") // 공백 기준으로 분리
        when(command[0]) {
            "push" -> que.add(command[1].toInt()) // 큐에 정수 추가
            "pop" -> sb.appendLine(if (que.isEmpty()) -1 else que.poll())
            "size" -> sb.appendLine(que.size)
            "empty" -> sb.appendLine(if (que.isEmpty()) 1 else 0)
            "front" -> sb.appendLine(if (que.isEmpty()) -1 else que.peek())
            "back" -> sb.appendLine(if (que.isEmpty()) -1 else que.peekLast())
        }
    }
    print(sb.toString())
}